### DISCUSSION TITLE
Autorange to include annotations

### DISCUSSION THREAD

**Discussion by colomb001** [jfree/jfreechart#379] [Q&A]
I've got a category plot on which I've added some annotations. Unfortunately, when setting the axis to auto-range, the annotations aren't taken into account and aren't visible. Is there a way to have auto-range consider annotations when calculating the visible chart?

**Comment by trashgod** [comment]
I'm not sure what you're seeing. Can you use the context menu or mouse gesture mentioned [here](https://stackoverflow.com/a/44151032/230513) restore the zoomed-out state?

**Comment by colomb001** [comment]
See example below. Using auto range, the annotation is not included. If you zoom out the range axis, you will see the annotation. My question: Is there a way to have annotations included when setting an axis to auto range.

`
import javax.swing.*;

import org.jfree.chart.ChartFactory;
import org.jfree.chart.ChartPanel;
import org.jfree.chart.JFreeChart;
import org.jfree.chart.annotations.CategoryPointerAnnotation;
import org.jfree.chart.axis.NumberAxis;
import org.jfree.chart.labels.StandardCategorySeriesLabelGenerator;
import org.jfree.chart.plot.CategoryPlot;
import org.jfree.chart.plot.PlotOrientation;
import org.jfree.chart.renderer.category.BarRenderer;
import org.jfree.data.category.CategoryDataset;
import org.jfree.data.category.DefaultCategoryDataset;

/**
 * A bar chart with only two bars - here the 'maximumBarWidth' attribute in the
 * renderer prevents the bars from getting too wide.
 */
public class BarChartDemo4 extends JFrame
{

    /**
     * Creates a new demo instance.
     *
     * @param title  the frame title.
     */
    public BarChartDemo4(String title) {
        super(title);
        JPanel chartPanel = createDemoPanel();
        chartPanel.setPreferredSize(new java.awt.Dimension(500, 270));
        setContentPane(chartPanel);
    }

    /**
     * Returns a sample dataset.
     *
     * @return The dataset.
     */
    private static CategoryDataset createDataset() {
        DefaultCategoryDataset dataset = new DefaultCategoryDataset();
        dataset.addValue(1.0, "First", "Category 1");
        dataset.addValue(5.0, "Second", "Category 1");
        return dataset;
    }

    /**
     * Creates a sample chart.
     *
     * @param dataset  the dataset.
     *
     * @return The chart.
     */
    private static JFreeChart createChart(CategoryDataset dataset) {

        // create the chart...
        JFreeChart chart = ChartFactory.createBarChart(
            "Bar Chart Demo 4",       // chart title
            null,                     // domain axis label
            "Value",                  // range axis label
            dataset,                  // data
            PlotOrientation.VERTICAL,
            true,                     // include legend
            true,                     // tooltips?
            false                     // URLs?
        );

        // get a reference to the plot for further customisation...
        CategoryPlot plot = (CategoryPlot) chart.getPlot();

        // set the range axis to display integers only...
        NumberAxis rangeAxis = (NumberAxis) plot.getRangeAxis();
        rangeAxis.setStandardTickUnits(NumberAxis.createIntegerTickUnits());

        // disable bar outlines...
        BarRenderer renderer = (BarRenderer) plot.getRenderer();
        renderer.setDrawBarOutline(false);

        // bars are capped at a maximum of 10 percent of the plot width
        renderer.setMaximumBarWidth(0.10);
        renderer.setLegendItemLabelGenerator(
            new StandardCategorySeriesLabelGenerator("{0} series"));

        plot.addAnnotation( new CategoryPointerAnnotation( "Test Annotation", "Category 1", 6, 30 ) );

        return chart;

    }

    /**
     * Creates a panel for the demo (used by SuperDemo.java).
     *
     * @return A panel.
     */
    public static JPanel createDemoPanel() {
        JFreeChart chart = createChart(createDataset());
        return new ChartPanel(chart);
    }

    /**
     * Starting point for the demonstration application.
     *
     * @param args  ignored.
     */
    public static void main(String[] args) {
        BarChartDemo4 demo = new BarChartDemo4(
            "JFreeChart: BarChartDemo4.java");
        demo.pack();
        demo.setVisible(true);
    }
}
`

**Comment by trashgod** [comment]
Several approaches come to mind:
- Invoke `rangeAxis.setRange(0, value)`, where `value` is sufficient to display the annotation.
- Invoke `plot.setRangePannable(true)` to let the user pan to the area of interest.
- Add controls like [this](https://stackoverflow.com/a/41544007/230513) to evoke the desired behavior.
